benchmark/parseandstatcompetition.cpp

```diff
@@ -105,7 +105,8 @@ void simdjson_recurse(stat_t &s, simdjson::dom::element element) {
 never_inline stat_t simdjson_compute_stats(const simdjson::padded_string &p) {
   stat_t s{};
   simdjson::dom::parser parser;
-  auto [doc, error] = parser.parse(p);
+  simdjson::dom::element doc;
```

Comment:
Here we see a bit of a negative: the user needs to know about simdjson::dom::element which wasn't the case before. 

Are we keeping the old API ( auto [doc, error] = parser.parse(p);)? I think that some users will prefer it.

I am "old school" and I prefer code that is explicitly typed. I don't like magic and surprises, but I can imagine people being annoyed at having to know the type that they need.

Reply:
(That's a minor point.)

Reply:
(And I understand the point about coding for C++11.)

Reply:
Yeah. I think there is likely a large intersection between users of error codes and explicit-type-lovers, but it's real :)

I find myself liking this because I'm *not* actually a huge fan of `auto` in the parser.parse() case. For me, the guideline is "omit the type iff it's obvious." The use of auto is for a sub-rule of that: rule "only write the type once per line, because it's obvious."

I think I'm a reasonable example of the user who hates typing unnecessary things (for many reasons). I went so far as to design a language and compiler partly around that hatred :)